analyzer/src/main/kotlin/managers/utils/NodeSupport.kt

```diff
@@ -135,6 +135,23 @@ fun readProxySettingsFromNpmRc(npmRc: String): ProtocolProxyMap {
     return map
 }
 
+const val PUBLIC_NPM_REGISTRY = "https://registry.npmjs.org"
+
+/**
+ * Return the npm registry defined in the provided [NPM configuration][npmRc].
+ */
+fun readNpmRegistryFromNpmRc(npmRc: String, default: String = PUBLIC_NPM_REGISTRY): String {
```

Comment:
Nit: This could probably just be named `readRegistryFromNpmRc` as it's clear that it's an *NPM* registry; also that nice reflect the "registry" string which is looked for.

Reply:
If this function returns "https://registry.npmjs.org", then it would be unclear whether that's because the `default` was returned, or because "https://registry.npmjs.org" was explicitly specified in the `.npmrc`. There might be cases where this is interesting to know; or phrased differently, it might be nice to be able to use this function to detect whether any custom registry is specified in `.npmrc`. As a result, I'd have a preference for this to return `null` instead of a default when no registry is defined in `.npmrc`, and let the caller handle any fallback.

Reply:
Ok, I understand